src/messaging/dotnet/src/Server/Server/MessageRouterServer.cs

```diff
@@ -322,11 +356,36 @@ public ValueTask ClientDisconnected(IClientConnection connection)
                 },
                 client);
 
-            return Task.CompletedTask;
+            await client.Connection.SendAsync(
+                new SubscribeResponse
+                {
+                    RequestId = message.RequestId
+                },
+                CancellationToken.None);
+
+            OnRequestStop(message);
         }
-        finally
+        catch (Exception exception)
         {
-            OnRequestStop(message);
+            if (_logger.IsEnabled(LogLevel.Debug))
+            {
+                _logger.LogDebug($"Exception thrown while handling subscription message: {0}.", exception);
```

Comment:
Are there not overloads that take an exception as a parameter?  Do we really need to create strings on the heap?